charts/kubearchive/templates/observability/grafana-lgtm-stack.yaml

```
@@ -0,0 +1,105 @@
+{{- if .Values.integrations.observability.enabled }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: grafana-lgtm
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: grafana-lgtm
+  template:
+    metadata:
+      labels:
+        app: grafana-lgtm
+    spec:
+      containers:
+        - name: grafana-lgtm
+          image: grafana/otel-lgtm:latest
+          ports:
+            - containerPort: 3000
+            - containerPort: 4317
+            - containerPort: 4318
+            - containerPort: 9411
+          volumeMounts:
+            - name: otelcol-config
+              mountPath: /otel-lgtm/otelcol-config.yaml
+              subPath: otelcol-config.yaml
+      volumes:
+        - name: otelcol-config
+          configMap:
+            name: otelcol-config
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: grafana-lgtm
+spec:
+  selector:
+    app: grafana-lgtm
+  ports:
+    - name: grafana
+      port: 3000
+    - name: otel-grpc
+      port: 4317
+    - name: otel-http
+      port: 4318
+    - name: zipkin
+      port: 9411
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: otelcol-config
+data:
+  otelcol-config.yaml: |
+    receivers:
+      otlp:
+        protocols:
+          grpc:
+            endpoint: 0.0.0.0:4317
+          http:
+            endpoint: 0.0.0.0:4318
+      prometheus/collector:
+        config:
+          scrape_configs:
+            - job_name: 'opentelemetry-collector'
+              static_configs:
+                - targets: ['localhost:8888']
+      zipkin:
+        endpoint: 0.0.0.0:9411
+
+    processors:
+      batch:
+
+    exporters:
+      otlphttp/metrics:
+        endpoint: http://localhost:9090/api/v1/otlp
+        tls:
+          insecure: true
+      otlphttp/traces:
+        endpoint: http://localhost:4418
+        tls:
+          insecure: true
+      otlphttp/logs:
+        endpoint: http://localhost:3100/otlp
+        tls:
+          insecure: true
+
+    service:
+      pipelines:
+        traces:
+          receivers: [otlp, zipkin]
+          processors: [batch]
+          exporters: [otlphttp/traces]
+        metrics:
+          receivers: [otlp,prometheus/collector]
+          processors: [batch]
+          exporters: [otlphttp/metrics]
+        logs:
+          receivers: [otlp]
+          processors: [batch]
+          exporters: [otlphttp/logs]
+
```

I would like to propose extracting the content of `otel-config.yaml` to a file.
I believe that having this kind of content directly embedded in the definition of the YAML file makes the chart and the configuration less readable (and there is no syntax highlight either).

The steps for extracting the file could be:

1.  Create a `grafana-otelcol-config.yaml` under `charts/kubearchive/conf/` with the contents of the `otelcol-config.yaml` in the configmap.
2.  Apply the following change on the configmap to make helm embed the content of the file again:
```suggestion
  otelcol-config.yaml: |-
    {{- tpl ($.Files.Get "conf/grafana-otelcol-config.yaml") . | nindent 4 }}
```

More information of this helm templating feature in [here](https://helm.sh/docs/chart_template_guide/accessing_files/#basic-example)

Using `{{ (.Files.Glob "conf/grafana-*").AsConfig | indent 2 }}` could work too, but then the file name has to match key in the ConfigMap

---

Done